core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -329,8 +360,15 @@ class GroupMetadataManager(brokerId: Int,
             responseCallback(commitStatus)
           }
 
-          group synchronized {
-            group.prepareOffsetCommit(offsetMetadata)
+          if (isTxnOffsetCommit) {
+            addProducerGroup(producerId, group.groupId)
```

Comment:
Mentioned offline, but adding this to make sure we don't forget about it. I think we need to make sure that we hold the group lock here so that we can keep the `pendingTransactionalOffsetCommits` collection in `GroupMetadata` in sync with `openGroupsForProducer`.